### PR TITLE
Added codecov integration (#436)

### PR DESCRIPTION
**Pull request by wcs1only**
# Description

Add codecov integration

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #436

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Review by youngbupark — APPROVED** [review]

**Review by tcnghia — APPROVED** [review]
Thanks for adding this

File: .github/workflows/sdk_build.yml [hunk 1]
```diff
@@ -27,17 +27,22 @@ jobs:
         uses: actions/setup-dotnet@v1
         with:
           dotnet-version: 3.1.301
+      - name: Test solution - release
+        run: |
+          unset GITHUB_ACTIONS #disable deterministic builds, just for test run. Deterministic builds break coverage for some reason
```
Comment (mukundansundar) [inline .github/workflows/sdk_build.yml:32]: is this only for dotnet or does this need to be added for other repos also?

**Review by mukundansundar — APPROVED** [review]

File: .github/workflows/sdk_build.yml (hunk 1 above)
Comment (wcs1only) [reply .github/workflows/sdk_build.yml:32]: There's an issue with the code coverage framework and deterministic builds
I tried the mitigation suggested in:
https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/DeterministicBuild.md
but I couldn't get it to work correctly. This was the simplest mitigation that I could get to work.

Comment (rynowak) [reply .github/workflows/sdk_build.yml:32]: I had some discussion about this offline. I'm going to take a look and see if I can figure out a better fix in a short amount of time. If not then we'll ship this.

**Review by rynowak — COMMENTED** [review]
Let's move forward with this for now.

The only downside to the workaround is a slower build by like 15 seconds.

Independently of this I'll follow up whether we *really* have deterministic builds working - this is something we care about because we need the supply chain to be verifiable.

**Pull request merged**